### PR TITLE
Update spectrum fitting utils to support wildcards in parameter name matching

### PR DESCRIPTION
**Pull request by joelvbernier**
Enhance parameter name matching to take wildcards.

### CODE REVIEW

**Comment by joelvbernier** [comment]
This needs a little more work, since it will _*only*_ supports wildcard strings now...  either I live with it or find a way to preserve (or choose by kwarg?) the old functionality using `__contains__` where the user could specify any substring w/o wildcards.

Any ideas here @psavery or @cjh1 or @donald-e-boyce ?

**Review by psavery — APPROVED** [review]
LGTM

File: hexrd/fitting/utils.py [hunk 1]
```diff
@@ -17,7 +19,8 @@ def _parameter_arg_constructor(pdict, pargs):
 
 
 def _extract_parameters_by_name(params, pname_root):
-    return [s for s in params.keys() if s.__contains__(pname_root)]
+    # return [s for s in params.keys() if s.__contains__(pname_root)]
+    return fnmatch.filter(list(params.keys()), pname_pattern)
```
Comment (psavery) [inline hexrd/fitting/utils.py:23]: @joelvbernier If you wanted to continue supporting substring matches, one option would be to just add `*` to the ends of the pattern, like so:

`f'*{pname_pattern}*'`

Then substrings would still match, and you can do `*` inside the string as well.

Alternatively, you could add an argument such as `match_substring=True` where, if it is true, then you add the `*` to the ends of the pattern.

Comment (joelvbernier) [reply hexrd/fitting/utils.py:23]: of course -- bracketing with '*' might be the most elegant solution :-)

**Comment by joelvbernier** [comment]
@psavery -- I think the previous commit is providing the behavior we want!  Nice one!

**Review by psavery — APPROVED** [review]
LGTM

**Pull request merged**